x-pack/metricbeat/module/prometheus/collector/histogram.go

```diff
@@ -51,9 +51,15 @@ func promHistogramToES(cc CounterCache, name string, labels common.MapStr, histo
 		}
 
 		// take count for this period (rate) + deacumulate
-		count := cc.RateUint64(name+labels.String()+fmt.Sprintf("%f", bucket.GetUpperBound()), bucket.GetCumulativeCount()) - sumCount
```

Comment:
@exekias I have seen with CockroachDB that this rate is sometimes zero when `sumCount` has already accumulated other counters, so the value overflows here and data fails to be ingested. Not sure why it happens, I want to investigate more, but I think that it may be that CockroachDB changes or at least adds bucket bounds on the go, not sure about the implications of this is this is the case.

By now I am workarounding this issue by explicitly handling zero-valued count rate.

Reply:
This fix could require a backport, right? If that's the case we will need another PR only for it. It would be interesting to know the root cause of this, adding buckets on the fly seems weird, but who knows

Reply:
Agree, I will move this fix to a different PR.

Reply:
PR open for this https://github.com/elastic/beats/pull/17753